src/ToricVarieties/cohomCalg/ShiftedCones/attributes.jl

```diff
@@ -0,0 +1,158 @@
+############################
+# Attributes
+############################
+
+@doc Markdown.doc"""
+    gens(sc::ShiftedCone)
+
+Return the generators of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> gens(sc)
+[1   2   3]
+[4   5   6]
+```
+"""
+function gens(sc::ShiftedCone)
+    return sc.gens::fmpz_mat
+end
+export gens
+
+
+@doc Markdown.doc"""
+    ngens(sc::ShiftedCone)
+
+Return the number of generators of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> ngens(sc)
+2
+```
+"""
+function ngens(sc::ShiftedCone)
+    return fmpz(nrows(sc.gens))::fmpz
+end
+export ngens
+
+
+@doc Markdown.doc"""
+    ambient_dim(sc::ShiftedCone)
+
+Return the ambient dimension of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> ambient_dim(sc)
+3
+```
+"""
+function ambient_dim(sc::ShiftedCone)
+    return fmpz(ncols(sc.gens))::fmpz
+end
+export ambient_dim
+
+
+@doc Markdown.doc"""
+    offset(sc::ShiftedCone)
+
+Return the offset of generators of the shifted cone `sc`.
+
+# Examples
+```jldoctest
+julia> ray_generators = [[1,2,3],[4,5,6]]
+2-element Vector{Vector{Int64}}:
+ [1, 2, 3]
+ [4, 5, 6]
+
+julia> shift = [-3,-2,-1]
+3-element Vector{Int64}:
+ -3
+ -2
+ -1
+
+julia> sc = ShiftedCone(ray_generators, shift)
+A shifted cone in ambient dimension 3
+
+julia> offset(sc)
+[-3   -2   -1]
+```
+"""
+function offset(sc::ShiftedCone)
+    return sc.offset::fmpz_mat
```

Comment:
See above... As usual I am not going to repeat my comments on all places where they might also apply :-)

Reply:
Right. I will check your comments throughout the `ToricVarieties`. Thank you!